pommerman/network/ion_client/__init__.py

```diff
@@ -0,0 +1,219 @@
+    from pommerman.constants import Item
+    if room:
+        room = str(ui.ask_string(Strings.room_code.value))
+    # Info: The Current Working Directory has to be added to path to import modules from it
+    path.append(getcwd())
+    if not (import_name and agent_class) and ui_en:
+        agent = import_module(ui.ask_string(Strings.match_import.value))
+        agent_class = ui.ask_string(Strings.match_class_name.value)
+    else:
+        agent = import_module(import_name)
+    if agent_class not in agent.__dir__():
+        if ui_en:
+            ui.fatal(Strings.error_invalid_class.value)
+        else:
+            raise Exception(constants.Exceptions.match_invalid_class.value)
+    agent = getattr(agent, agent_class)
+    if getattr(agent, "act") and ui_en:
+        ui.info(ui.green, Strings.match_agent_success.value)
+    agent = agent()
+    if ui_en:
+        ui.info(ui.yellow, Strings.server_comm.value)
+    try:
+        network.join_list(room)
+    except Exception as e:
+        if ui_en:
+            ui.fatal(e)
+        raise e
+    if ui_en:
+        ui.info(Strings.match_variant.value, ui.yellow, network.mode)
+        ui.info(ui.yellow, Strings.match_wait.value)
+    try:
+        network.waitm()
+    except Exception as e:
+        if ui_en:
+            ui.fatal(e)
+        raise e
+    if ui_en:
+        ui.info(Strings.match_run.value, "#" + network.match_id)
+    while (True):
+        try:
+            match_obj = network.match_get()
+        except Exception as e:
+            if ui_en:
+                ui.fatal(e)
+            raise e
+        if match_obj[0] is 0:
+            action = agent.act(match_obj[1], Discrete(6))
+            try:
+                network.send_move(action, match_obj[2])
+            except Exception as e:
+                if ui_en:
+                    ui.fatal(e)
+                raise e
+        elif match_obj[0] is 2:
+            agent.episode_end(match_obj[1])
+            if match_obj[1] == 1:
+                ui.info(Strings.match_won.value)
+            if match_obj[1] == -1:
+                ui.info(Strings.match_loss_draw.value)
+            ui.info(
+                Strings.match_agent.value, ui.yellow,
+                Item(match_obj[2]).name
+            )
+            break
+    ui.info(Strings.match_replay.value, ui.yellow, network.match_id)
+    if ui.ask_yes_no(Strings.match_ask_replay.value):
+        replay(network, network.match_id)
+    else:
+        intent(network)
+
+
+def replay(network, id=False, ui_en=True):
+    """Arguments:  
+    * network: An `network`(pommerman.network.ion_client.network) object  
+    * id: The ID of a match to be played. If False, the user is prompted about \
+it.  
+    * ui_en: If the UI is enabled or disabled (This also controls if exception are\
+raised or not)  
+    Description: This replays a particular match"""
+    if not id and ui_en:
+        id = str(ui.ask_string(Strings.replay_prompt.value))
+        if id is None:
```

Comment:
I don't think that this will catch anything because str(None) = 'None' != None.

Reply:
Well, this comes down to the specificities of the library. `ui` returns `None` incase the input field was blank.

Reply:
If `ui` returns `None`, then what is `str(None)` and does it ever `= None`? 
More particularly, can your 'id' ever be 'None'?